torch/csrc/autograd/python_function.h

```diff
@@ -88,6 +88,9 @@ struct THPFunction {
     // pass.  Set by Python with 'mark_dirty'.  If nullptr, no tensors were
     // modified inplace.
     PyObject *dirty_tensors;
+    // Python boolean indicating whether to materialize undefined output grad
+    // tensors into tensors full of zeros. Set by Python with 'set_materialize_grads'.
+    PyObject *materialize_grads;
```

Comment:
Maybe you want to save this as a c++ bool no?
This will avoid any issue with the refcounting.
It will allow you to have a good place to do proper type checking on the argument.
It will be much more performant for users that don't use this feature (comparing a bool vs two PyObjects).
It will also make sure you don't need the `__init__()` on the python side which looks dangerous here.
You can simply register `set_materialize_grads` as a method directly and make it save `arg == Py_True`.